api/reviews_api.py

```diff
@@ -163,6 +163,52 @@ def validate_assignees(self, assignees, fe, gate):
         self.abort(400, 'Assignee is not a reviewer')
 
 
+class PendingGatesAPI(basehandlers.APIHandler):
+
+  def get_stage_ids_of_gates_pending_my_approval(self) -> list[int]:
+    """Return a list of stage_id needing approval by current user."""
+    user = self.get_current_user()
+    if not user:
+      return []
+
+    approvable_gate_types = approval_defs.fields_approvable_by(user)
+    if not approvable_gate_types:
+      logging.info('User has no approvable_gate_types')
+      return []
+    query = Gate.query(
+        Gate.state.IN(Gate.PENDING_STATES),
+        Gate.gate_type.IN(approvable_gate_types))
+    future_stage_ids = query.fetch_async(projection=['stage_id'])
```

Comment:
Could we call `get_result()` here at the end of the method in order to keep the return type consistently list[int], or does that eliminate the purpose of doing this asynchronously?

Reply:
Right, that would lose the benefit.